### PR TITLE
Switch to using external ECR credential provider for k8s 1.27

### PR DESCRIPTION
**Pull request by olemarkus**

### CODE REVIEW

File: nodeup/pkg/model/kubelet.go [hunk 1]
```diff
@@ -413,6 +434,56 @@ func (b *KubeletBuilder) usesContainerizedMounter() bool {
 	}
 }
 
+// addECRCP installs the ECR credential provider
+func (b *KubeletBuilder) addECCRCP(c *fi.NodeupModelBuilderContext) error {
```
Comment (hakman) [inline nodeup/pkg/model/kubelet.go:438]:
```suggestion
func (b *KubeletBuilder) addECRCP(c *fi.NodeupModelBuilderContext) error {
```

File: pkg/apis/kops/cluster.go [hunk 2]
```diff
@@ -220,6 +220,9 @@ type AWSSpec struct {
 	// Spotinst cloud-config specs
 	SpotinstProduct     *string `json:"spotinstProduct,omitempty"`
 	SpotinstOrientation *string `json:"spotinstOrientation,omitempty"`
+
+	// BinaryLocation is the location of the AWS cloud provider binaries.
+	BinaryLocation *string `json:"binaryLocation,omitempty"`
```
Comment (hakman) [inline pkg/apis/kops/cluster.go:225]: `BinaryLocation` seems a little vague to me.

Comment (olemarkus) [reply pkg/apis/kops/cluster.go:225]: Yeah I know. I hate it. But I couldn't find a good name for this.

FTR: the motivation behind this one is to use the PR/staging bucket in the AWS cloud provider e2e tests.

File: upup/pkg/fi/cloudup/apply_cluster.go [hunk 3]
```diff
@@ -1056,6 +1056,23 @@ func (c *ApplyClusterCmd) addFileAssets(assetBuilder *assets.AssetBuilder) error
 			c.Assets[arch] = append(c.Assets[arch], mirrors.BuildMirroredAsset(u, hash))
 		}
 
+		if c.Cluster.UsesExternalECRCredentialsProvider() {
+			binaryLocation := c.Cluster.Spec.CloudProvider.AWS.BinaryLocation
+			if binaryLocation == nil {
+				binaryLocation = fi.PtrTo("https://artifacts.k8s.io/binaries/cloud-provider-aws/v1.27.1")
```
Comment (hakman) [inline upup/pkg/fi/cloudup/apply_cluster.go:1062]: Should `v1.27.1` be something like AWS CCM version? Not sure if we have access to that here.

Comment (olemarkus) [reply upup/pkg/fi/cloudup/apply_cluster.go:1062]: Its a good catch. I was thinking of a helper method here where we feed in the K8s version and returns the cloud provider version. And move the case statement from CCM into that function. But I thought there wasn't a rush to do this right now.

Comment (hakman) [reply upup/pkg/fi/cloudup/apply_cluster.go:1062]: No rush.

**Review by hakman — APPROVED** [review]

**Pull request merged**